src/main/webapp/app/core/main/services/productService.js

```diff
@@ -3,7 +3,7 @@
 (function () {
     var app = angular.module("efApp");
 
-    app.service("ProductService", function ProductService($http, BASE_SERVER_URL, AuthService) {
+    app.service("ProductService", function ProductService($http, $q, BASE_SERVER_URL, AuthService) {
```

Comment:
Always add new injections in the end of the injections to don't look like that you have created a new line

Reply:
just to say, you can let It this way